### ISSUE TITLE
Custom Gateway [Feature Request]

### ISSUE THREAD

**Issue by heberq1978** [FOSSBilling/FOSSBilling#2042]
I would like the customized payment to have keys to show the customer the amount to be paid, plan purchased, expiration and expiration date, customer ID, among other information that could be used together with Html + JavaScript.

Example:

[client_name] => name client
[client_id] => id client
[plan] => purchased plan
[plan_validate] => plan validity in days
[plan_invoice] => invoice due date
[]plan_price] => payment amount

And could insert other keys, so we can generate invoices using HTML + javascript via the Custom Gateway.

Thank You.

**Comment by BelleNottelling** [comment]
This is also a duplicate of #694

**Comment by heberq1978** [comment]
Thank you Belle, and I'm sorry for opening the same ticket more than once.
Now taking advantage, you could remember us Brazilians, as we are without a gateway to work with Foss. Not just from Brazilians, but from Latin America, who have Mercado Pago as their standard.

**Comment by BelleNottelling** [comment]
> Thank you Belle, and I'm sorry for opening the same ticket more than once.

It's not really a big deal.

Also, I'm not sure if you saw, but I realized there's already some variable functionality and I gave steps on how to view what's available here: https://github.com/FOSSBilling/FOSSBilling/issues/694#issuecomment-1879206143

> Now taking advantage, you could remember us Brazilians, as we are without a gateway to work with Foss. Not just from Brazilians, but from Latin America, who have Mercado Pago as their standard.

I'm aware of that, but there's a lot of work to do to get FOSSBilling to what we consider a production-ready state and a lot of functionality is planned to be refactored, rewritten, or removed to get the core software to a more modern, stable, flexible, and secure state. These changes to the core software aren't small and they can take a lot of time to be completed. There are only a few people who actively contribute to the project and we aren't paid to do this work.

Some of us have given 100s if not 1000s of hours of our free time to continue moving the project forward and it's not really practical nor fair to think that the few of us will build every integration for every user for no cost.

If somebody needs an integration for FOSSBilling so they they can sell a product, they should either be willing to pay someone to create that for them or learn PHP and implement it themselves. At the end of the day FOSSBilling is a completely free and open source tool to help businesses make money and I think it's very reasonable to expect a business to be willing to directly support their ability to function as a business either through a time investment or actual monetary spending.
